// @msgpack/msgpack/dist/index.js
"use strict";
// Main Functions:
Object.defineProperty(exports, "__esModule", { value: true });
var encode_1 = require("./encode");
exports.encode = encode_1.encode;
var decode_1 = require("./decode");
exports.decode = decode_1.decode;
var decodeAsync_1 = require("./decodeAsync");
exports.decodeAsync = decodeAsync_1.decodeAsync;
exports.decodeArrayStream = decodeAsync_1.decodeArrayStream;
exports.decodeStream = decodeAsync_1.decodeStream;
/**
 * @experimental `Decoder` is exported for experimental use.
 */
var Decoder_1 = require("./Decoder");
exports.Decoder = Decoder_1.Decoder;
/**
 * @experimental `Encoder` is exported for experimental use.
 */
var Encoder_1 = require("./Encoder");
exports.Encoder = Encoder_1.Encoder;
// Utilitiies for Extension Types:
var ExtensionCodec_1 = require("./ExtensionCodec");
exports.ExtensionCodec = ExtensionCodec_1.ExtensionCodec;
var ExtData_1 = require("./ExtData");
exports.ExtData = ExtData_1.ExtData;
var timestamp_1 = require("./timestamp");
exports.EXT_TIMESTAMP = timestamp_1.EXT_TIMESTAMP;
exports.encodeDateToTimeSpec = timestamp_1.encodeDateToTimeSpec;
exports.encodeTimeSpecToTimestamp = timestamp_1.encodeTimeSpecToTimestamp;
exports.decodeTimestampToTimeSpec = timestamp_1.decodeTimestampToTimeSpec;
exports.encodeTimestampExtension = timestamp_1.encodeTimestampExtension;
exports.decodeTimestampExtension = timestamp_1.decodeTimestampExtension;
